### PR TITLE
[viessmann] add command option and state description

### PR DESCRIPTION
**Pull request by rogrun**
This PR adds automatically generated command options depending on the provided commands from the API

### CODE REVIEW

**Comment by cweitkamp** [comment]
One comment. You do not need both a `CommandDescription` and a `StateDescription` provider for the same channel. If you have a channel exposing a state use the later one. It decodes states to commands with the same meaning. If you have a state-less Channel which use the `CommandDescriptionProvider`.

**Comment by rogrun** [comment]
Good to know, I will change that and give feedback on it

File: bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/ViessmannDynamicCommandDescriptionProvider.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * Dynamic provider of command options.
+ *
+ * @author Ronny Grun - Initial contribution
+ */
+@Component(service = { DynamicCommandDescriptionProvider.class, ViessmannDynamicCommandDescriptionProvider.class })
+@NonNullByDefault
+public class ViessmannDynamicCommandDescriptionProvider extends BaseDynamicCommandDescriptionProvider {
```
Comment (cweitkamp) [inline bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/ViessmannDynamicCommandDescriptionProvider.java]: Please correct me if I am wrong: you can remove the entire class.

**Review by cweitkamp — CHANGES_REQUESTED** [review]

File: bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/ViessmannDynamicCommandDescriptionProvider.java (hunk 1 above)
Comment (rogrun) [reply bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/ViessmannDynamicCommandDescriptionProvider.java]: yes of course :) I'll remove this class

File: bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/DeviceHandler.java [hunk 2]
```diff
@@ -60,6 +63,7 @@
  * @author Ronny Grun - Initial contribution
  */
 @NonNullByDefault
+@SuppressWarnings("null")
```
Comment (J-N-K) [inline bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/DeviceHandler.java]: Why do you suppress this warning?

Comment (rogrun) [reply bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/DeviceHandler.java]: I use VS Code as an editor. Recently I've been getting warnings like this one:
`Null type safety (type annotations): The expression of type 'Logger' needs unchecked conversion to conform to '@NonNull Logger'`
I solved it by suppress this. I don't know if it is the right way. Should I change this?

**Review by cweitkamp — DISMISSED** [review]

**Review by cweitkamp — APPROVED** [review]

**Review by J-N-K — APPROVED** [review]

**Pull request merged**